From 543bd115e1434681064718be6d6a9e04f6886e27 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <Michal@Hrusecky.net>
Date: Mon, 19 Feb 2018 22:02:58 +0100
Subject: [PATCH] kernel: Include kernel magic in all kmods

Kernel modules can greatly depend on configuration of the kernel and might not
work with different kernel, so always put kernel magic as part of the package
version.

Also install modules into directories versioned by the kernel magic.

Signed-off-by: Michal Hrusecky <Michal@Hrusecky.net>
---
 include/kernel-defaults.mk | 2 ++
 include/kernel.mk          | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/kernel-defaults.mk b/include/kernel-defaults.mk
index cc1e236..78f002b 100644
--- a/include/kernel-defaults.mk
+++ b/include/kernel-defaults.mk
@@ -106,6 +106,8 @@ define Kernel/Configure/Default
 	}
 	$(_SINGLE) [ -d $(LINUX_DIR)/user_headers ] || $(KERNEL_MAKE) INSTALL_HDR_PATH=$(LINUX_DIR)/user_headers headers_install
 	grep '=[ym]' $(LINUX_DIR)/.config.set | LC_ALL=C sort | mkhash md5 > $(LINUX_DIR)/.vermagic
+	sed -i '/^CONFIG_LOCALVERSION=.*/ d' $(LINUX_DIR)/.config
+	echo CONFIG_LOCALVERSION=\""-`cat $(LINUX_DIR)/.vermagic`"\"  >> $(LINUX_DIR)/.config;
 endef
 
 define Kernel/Configure/Initramfs
diff --git a/include/kernel.mk b/include/kernel.mk
index 97d2366..93be63d 100644
--- a/include/kernel.mk
+++ b/include/kernel.mk
@@ -208,8 +208,8 @@ define KernelPackage
     SECTION:=kernel
     CATEGORY:=Kernel modules
     DESCRIPTION:=$(DESCRIPTION)
+    VERSION:=$(LINUX_VERSION)$(if $(PKG_VERSION),+$(PKG_VERSION))-$(if $(PKG_RELEASE),$(PKG_RELEASE),$(LINUX_RELEASE))-$(LINUX_VERMAGIC)
     EXTRA_DEPENDS:=kernel (=$(LINUX_VERSION)-$(LINUX_RELEASE)-$(LINUX_VERMAGIC))
-    VERSION:=$(LINUX_VERSION)$(if $(PKG_VERSION),+$(PKG_VERSION))-$(if $(PKG_RELEASE),$(PKG_RELEASE),$(LINUX_RELEASE))
     PKGFLAGS:=$(PKGFLAGS)
     $(call KernelPackage/$(1))
     $(call KernelPackage/$(1)/$(BOARD))
-- 
2.24.0

